File: yht0827/2월/2월 24일/택배상자.java
import java.util.*;

class Solution {

	public int solution(int[] order) {
		int answer = 0;
		Stack<Integer> stack = new Stack<>();
		int size = order.length;
		int index = 0;
		for (int i = 1; i <= size; i++) {
			if (order[index] != i) {
				stack.push(i);
			} else {
				index++;
				answer++;
			}
			while (!stack.isEmpty() && stack.peek() == order[index]) {
				stack.pop();
				index++;
				answer++;
			}
		}
		return answer;
	}
}

File: jeong-hyuk/2월24일/택배상자.py
def solution(order):    
  assist = []    
  i = 1    
  cnt = 0     
  while i != len(order)+1:        
    assist.append(i)        
    while assist and assist[-1] == order[cnt]:            
      cnt += 1            
      assist.pop()                    
      i += 1    
  return cnt
